transactions/transactions.py

```diff
@@ -36,6 +49,92 @@ async def genericAnnounce(self, ctx, *, message):
         except KeyError:
             await ctx.send(":x: Transaction log channel not set")
 
+    @commands.command(aliases=['makeFA'])
+    @commands.guild_only()
+    @checks.admin_or_permissions(manage_guild=True)
+    async def expireContracts(self, ctx: commands.Context, *userList):
+        """Displays each member that can be found from the userList a Free Agent in their respective tier"""
+        empty = True
+        free_agents = 0
+        notFound = 0
+        message = ""
+        fa_role = self.team_manager_cog._find_role_by_name(ctx, "Free Agent")
+        league_role = self.team_manager_cog._find_role_by_name(ctx, "League")
+
+        roles_to_remove = [
+            self.team_manager_cog._find_role_by_name(ctx, "Draft Eligible"),
+            self.team_manager_cog._find_role_by_name(ctx, self.PERM_FA_ROLE),
+            self.team_manager_cog._find_role_by_name(ctx, "Former Player")
+        ]
+
+        trans_channel : discord.TextChannel = await self._trans_channel(ctx)
+
+        for user in userList:
+            try:
+                member : discord.Member = await commands.MemberConverter().convert(ctx, user)
+            except Exception as e:
+                await ctx.send(f"Error: {e}")
+                if notFound == 0:
+                    message += "Couldn't find:\n"
+                message += "{0}\n".format(user)
+                notFound += 1
+            
+            # Process Contract expiration
+            # For each user in guild
+            if member in ctx.guild.members:
+
+                # prep roles to remove
+                franchise_role = self.team_manager_cog.get_current_franchise_role(member)
+                removable_roles = [franchise_role] if franchise_role else []
+                for role in roles_to_remove:
+                    if role in member.roles:
+                        removable_roles.append(role)
+                
+                # prep roles to add
+                tier_role = await self.team_manager_cog.get_current_tier_role(ctx, member)
+
+                if tier_role:
+                    tier_fa_role: discord.Role = self.team_manager_cog._find_role_by_name(ctx, tier_role.name + "FA")
+                    add_roles = [league_role, fa_role, tier_fa_role]
+                else:
+                    add_roles = [league_role, fa_role]
+
+                # get team/franchise info before role removal
+                team = (await self.team_manager_cog.teams_for_user(ctx, member))[0]
+                gm_name = self.team_manager_cog._get_gm_name(franchise_role)
+                # franchise_name = self.team_manager_cog.get_franchise_name_from_role(franchise_role)
+                gm: discord.Member = self.team_manager_cog._find_member_by_name(ctx, gm_name)
+
+                # performs role updates
+                await member.remove_roles(*removable_roles)
+                await member.add_roles(*add_roles)
```

Comment:
can these be done concurrently?

```suggestion
                await asyncio.gather(member.remove_roles(*removable_roles), member.add_roles(*add_roles))
```

Reply:
Possibly. What does `.gather` do?

Reply:
Gave it a quick test. It doesn't really work out, I ran into some of the same issues where getting access to team/franchise info didn't work properly. So its better for the roles to be removed then added first, awaiting that behavior.